platform/ios/src/MGLMapView.mm

```diff
@@ -5997,7 +5997,8 @@ - (CGPoint)userLocationAnnotationViewCenter
     
     if ([self.delegate respondsToSelector:@selector(mapViewUserLocationAnchorPoint:)])
     {
-        center = [self.delegate mapViewUserLocationAnchorPoint:self];
+        CGPoint anchorPoint = [self.delegate mapViewUserLocationAnchorPoint:self];
+        center = CGPointMake((contentFrame.origin.x + anchorPoint.x), (contentFrame.origin.y + anchorPoint.y));
```

Comment:
By this point, `contentFrame` has already been inset by `-edgePaddingForFollowingWithCourse`, which might not be quite correct. That method does account for the size of the puck, which is probably a good thing, but it also applies `MGLUserLocationAnnotationViewInset`.